### ISSUE TITLE
debian package: run tests

### ISSUE THREAD

**Issue by CastixGitHub** [i3-workspace-names-daemon/i3-workspace-names-daemon#21]
tests are not ran when building the deb file

**Comment by FloydOtsoko** [comment]
as far as I understand the situation tests are run whenever you push to any branch.

Packages are currently built whenever a tag in the style of 'v*' is pushed.
So technically one could build based on a commit which did not pass the tests, but it would be impossible to build on a commit which has not been tested at all.

In order to have the package task rely on the test task (aka. only run, when tests have passed) we would need to trigger the package building via the test task, after it has succeeded.

As far as I understand it, this would either require a separate test run, which is triggered by the tags, or we would build a new package whenever the test suite succeeds. Which imho would lead to a useless number of packages.

Tl;dr: contributors need to use their brain, to decide which commit to pack. If they do, everything should have been tested before building starts.

**Comment by CastixGitHub** [comment]
This issue is about building the debian package locally instead of relying on github
Sorry about lack of a proper description

**Comment by FloydOtsoko** [comment]
I have to admit I have forgotten almost everything I did when I created the deb packaging.

As of now one can run ` debuild --no-tgz-check -us -uc`  in the root directory of the repo, and it will build the package locally.

One can install it and test it. If that is what you meant...

**Comment by CastixGitHub** [comment]
What I mean is that usually test ran on a clone give different results compared to the ones ran from a distribution package
-- errors like missing to add a new file into MANIFEST.in
So you can also run the test suite when you build the deb
- let's say we add a check on a particular font version

Doing so eases the development of debian based distributions

This is not an issue until someone points it out, I tried to act eager opening the issue
